script.js

```diff
@@ -2,11 +2,71 @@
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  console.log(allEpisodes);
 }
-
+ const rootElem = document.getElementById("root");
+// show all episod lists in main page
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+   episodeList.forEach(episod=>{
+    const episodContainer= document.createElement("div");
+    const title=document.createElement("h3");
+    const image=document.createElement("img");
+    const summary=document.createElement("p");
+    episod.number < 10 && episod.season < 10
+      ? (title.textContent = `${episod.name}-S0${episod.number}E0${episod.season}`)
+      : (title.textContent = `${episod.name}-S${episod.number}E${episod.season}`);
+    image.src=`${episod.image.medium}`
+    summary.textContent = `${episod.summary}`.replace(/(<([^>]+)>)/gi, "");
+    episodContainer.appendChild(title);
+    episodContainer.appendChild(image);
+    episodContainer.appendChild(summary);
+    rootElem.appendChild(episodContainer);
+  });
 }
-
-window.onload = setup;
+  // search section
+  const form=document.getElementById("form");
+  const searchInput = document.getElementById("search");
+  let episodeList=getAllEpisodes();
+  searchInput.addEventListener("keyup",(e)=>{
+    let value =e.target.value;
+    let html = "";
+    episodeList.forEach((element)=>{
+      if (element.name.toLowerCase().includes(value) || element.summary.toLowerCase().includes(value)) {
+         html += `<div>
+        <h3>${element.name}</h3>
+        <img src=${element.image.medium}>
+        <p>${element.summary}</p>
+        </div>`;
+      }
+    })
```

Comment:
this works but you already have a reusable function (makePageForEpisodes) that you could use here. You can filter the episodes, save to a variable and then pass that variable into your makePageForEpisodes function. Just remember to clear the html you already have first!